packages/fuels-contract/src/contract.rs

```diff
@@ -156,8 +160,10 @@ impl Contract {
         binary_filepath: &str,
         wallet: &LocalWallet,
         params: TxParameters,
+        storage_slots: Vec<StorageSlot>,
```

Comment:
Instead of taking in a mandatory vector (which from the tests is empty 100% of the time), why not change `deploy_with_salt` to `deploy_with_parameters` (or something like that), that offloads all the complex parameter choosing to this advanced function. Then the vanilla `deploy` can be as simple as possible.

The simplest use case is users will want to deploy a contract with initial storage slots set automatically from a file, with a salt of zero. Anything else is advanced.